opm/utility/ECLSaturationFunc.cpp

```diff
@@ -2474,13 +2474,17 @@ pcowCurve(const ECLInitFileData& init,
     auto pc = this->wat_->pcow(satnum - 1, sw_lookup);
 
     if (enableVerticalSFScaling(scaling)) {
-        const auto fvals = SatFunc::CreateEPS::Vertical::
+        auto fvals = SatFunc::CreateEPS::Vertical::
             unscaledFunctionValues(init, gridID, satnum, *this->tep_, opt,
                 [this](const int satreg, const double s)
             {
                 return this->wat_->pcow(satreg, std::vector<double>{s}).front();
             });
 
+        // Special case treatment of PCOW.  Maximum function value at
+        // minimum Sw (here known as "disp").
+        fvals.max.val = std::max(fvals.disp.val, fvals.max.val);
```

Comment:
Suggest setting fvals.max = fvals.disp to also get the saturation correct, when fvals.disp.val > fvals.max.val.